org.eclipse.m2e.core.ui/src/org/eclipse/m2e/core/ui/internal/preferences/CentralSearchPreferencePage.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Sonatype Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *      Sonatype Inc. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.m2e.core.ui.internal.preferences;
+
+import org.eclipse.jface.preference.FieldEditorPreferencePage;
+import org.eclipse.jface.preference.StringFieldEditor;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPreferencePage;
+
+import org.eclipse.m2e.core.internal.preferences.MavenPreferenceConstants;
+import org.eclipse.m2e.core.ui.internal.M2EUIPluginActivator;
+import org.eclipse.m2e.core.ui.internal.Messages;
+
+
+/*
+ * @author Matthew Piggott
+ * @since 1.17.0
+ */
+public class CentralSearchPreferencePage extends FieldEditorPreferencePage implements IWorkbenchPreferencePage {
```

Comment:
Why isn't this preference page inside the search.central bundle?

Reply:
> Why isn't this preference page inside the search.central bundle?

At the moment there is also the org.eclipse.m2e.sourcelookup code where it may make sense to replace the existing code (which also talks to S.M.O). Once upon a time there was also the thought that some of the m2e bundles might be headless use cases.

Reply:
so there should be a search.central.ui bundle for that page. Or -unless there is already a compelling use-case for headless bundles-, just make it really all about UI and get rid of all the complications brought by the attempt to work headless.